src/android/nl/xservices/plugins/Calendar.java

```diff
@@ -213,7 +213,8 @@ public void onRequestPermissionResult(int requestCode, String[] permissions, int
 
   private void openCalendarLegacy(JSONArray args) {
     try {
-      final Long millis = args.getJSONObject(0).optLong("date");
+      long provided = args.getJSONObject(0).optLong("date", -1);
+      final long millis = provided < 0 ? new Date().getTime() : provided;
```

Comment:
Was there any problem with the boxed wrapper for `long`?

Reply:
No, it was just not necessary since the output of optLong is not optional (can't be null). Using Long threw a warning in Android Studio.